### ISSUE TITLE
411 Length Required

### ISSUE THREAD

**Issue by neojski** [chbrown/flickr-with-uploads#5]
Whenever I try to upload an image I get:

``` xml
<html>
<head><title>411 Length Required</title></head>
<body bgcolor="white">
<center><h1>411 Length Required</h1></center>
<hr><center>nginx/1.2.0</center>
</body>
</html>
```

and in fact `content-length` header is not sent. Maybe flickr started to require this at some point.

**Comment by neojski** [comment]
Basically as a workaround I did:

``` js
form.getLength(function(err, length){
    var headers = form.getHeaders({'content-length': length});

    var payload = {host: api_url_parts.host, path: '/services/'+ self.method+'/?format=json', headers: headers, method: 'POST'};
    var req = http.request(payload, function(res) { self.handleResponseStream(res); });
    form.pipe(req);
  });
```

If you're satisfied with this solution I can make a PR.

**Comment by chbrown** [comment]
Cool. And that still works with Flickr's weird flavor of OAuth signing? I'm actually making some changes to my flickr-backup package right now, so I can just paste in that change while I'm in flickr development mode.

**Comment by neojski** [comment]
It works for mew with OAuth signing. To be honest I'm not yet OAuth master but I think that headers don't change the signature. By the way - how do you get your oauth_token and oauth_token_secret? I currenlty have them from some other python library but is it possible to get them painlessly using js oauth?

I just did (#6) a PR for this so hope it helps even though you're in develpment mode :-)

**Comment by chbrown** [comment]
Fixed as of v1.0.3, thanks for the pointer to form-data's `.submit()` call.
